aws/resource_aws_service_discovery_http_namespace.go

```diff
@@ -0,0 +1,132 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsServiceDiscoveryHttpNamespace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceDiscoveryHttpNamespaceCreate,
+		Read:   resourceAwsServiceDiscoveryHttpNamespaceRead,
+		Delete: resourceAwsServiceDiscoveryHttpNamespaceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateServiceDiscoveryHttpNamespaceName,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceDiscoveryHttpNamespaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sdconn
+
+	name := d.Get("name").(string)
+	// The CreatorRequestId has a limit of 64 bytes
+	var requestId string
+	if len(name) > (64 - resource.UniqueIDSuffixLength) {
+		requestId = resource.PrefixedUniqueId(name[0:(64 - resource.UniqueIDSuffixLength - 1)])
+	} else {
+		requestId = resource.PrefixedUniqueId(name)
+	}
+
+	input := &servicediscovery.CreateHttpNamespaceInput{
+		Name:             aws.String(name),
+		CreatorRequestId: aws.String(requestId),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreateHttpNamespace(input)
+	if err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{servicediscovery.OperationStatusSubmitted, servicediscovery.OperationStatusPending},
+		Target:  []string{servicediscovery.OperationStatusSuccess},
+		Refresh: servicediscoveryOperationRefreshStatusFunc(conn, *resp.OperationId),
+		Timeout: 5 * time.Minute,
+	}
+
+	opresp, err := stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*opresp.(*servicediscovery.GetOperationOutput).Operation.Targets["NAMESPACE"])
```

Comment:
Normally we'd ask for `d.SetId()` to be right after the creation and before the waiter to prevent creating potentially orphaned resources, but since the API doesn't return the namespace ID until you read the operation in the waiter, this is okay. 👍